tests/unittests/UIKit/NSParagraphStyleTests.mm

```diff
@@ -0,0 +1,87 @@
+    NSMutableParagraphStyle* style = [[NSMutableParagraphStyle new] autorelease];
+    style.alignment = NSTextAlignmentCenter;
+    style.lineSpacing = 10.0;
+    woc::unique_cf<CTParagraphStyleRef> convertedStyle{[style _createCTParagraphStyle] };
+    CTTextAlignment alignment;
+    EXPECT_TRUE(CTParagraphStyleGetValueForSpecifier(convertedStyle.get(),
+                                                     kCTParagraphStyleSpecifierAlignment,
+                                                     sizeof(CTTextAlignment),
+                                                     &alignment));
+    EXPECT_EQ(kCTCenterTextAlignment, alignment);
+
+    CGFloat lineSpacing;
+    EXPECT_TRUE(
+        CTParagraphStyleGetValueForSpecifier(convertedStyle.get(), kCTParagraphStyleSpecifierLineSpacing, sizeof(CGFloat), &lineSpacing));
+    EXPECT_NEAR(10.0, lineSpacing, c_errorDelta);
+
+    CGFloat maximumLineHeight = 123545.67;
+    EXPECT_TRUE(CTParagraphStyleGetValueForSpecifier(convertedStyle.get(),
+                                                     kCTParagraphStyleSpecifierMaximumLineHeight,
+                                                     sizeof(CGFloat),
+                                                     &maximumLineHeight));
+    EXPECT_NEAR(0.0, maximumLineHeight, c_errorDelta);
+
+    NSParagraphStyle* copiedStyle = [[style copy] autorelease];
+    convertedStyle.reset([copiedStyle _createCTParagraphStyle]);
+    EXPECT_TRUE(CTParagraphStyleGetValueForSpecifier(convertedStyle.get(),
+                                                     kCTParagraphStyleSpecifierAlignment,
+                                                     sizeof(CTTextAlignment),
+                                                     &alignment));
+    EXPECT_EQ(kCTCenterTextAlignment, alignment);
+
+    EXPECT_TRUE(
+        CTParagraphStyleGetValueForSpecifier(convertedStyle.get(), kCTParagraphStyleSpecifierLineSpacing, sizeof(CGFloat), &lineSpacing));
+    EXPECT_NEAR(10.0, lineSpacing, c_errorDelta);
+
+    EXPECT_TRUE(CTParagraphStyleGetValueForSpecifier(convertedStyle.get(),
+                                                     kCTParagraphStyleSpecifierMaximumLineHeight,
+                                                     sizeof(CGFloat),
+                                                     &maximumLineHeight));
+    EXPECT_NEAR(0.0, maximumLineHeight, c_errorDelta);
+}
+
+TEST(NSParagraphStyle, CopyShouldBeImmutable) {
+    NSMutableParagraphStyle* style = [[NSMutableParagraphStyle new] autorelease];
+    style.alignment = NSTextAlignmentCenter;
+    style.lineSpacing = 10.0;
+    id copiedStyle = [[style copy] autorelease];
+    EXPECT_ANY_THROW([copiedStyle setParagraphStyle:style]);
+
+    style.lineSpacing = 20.0;
+    EXPECT_NEAR(10.0, [copiedStyle lineSpacing], c_errorDelta);
+}
+
+TEST(NSParagraphStyle, DefaultShouldReturnSameInstance) {
+    NSParagraphStyle* first = [NSParagraphStyle defaultParagraphStyle];
```

Comment:
nit: i don't know whether this should be tested; we don't want to capture platform version behavioural changes